src/main/java/com/synopsys/integration/detect/tool/signaturescanner/BlackDuckSignatureScanner.java

```diff
@@ -163,8 +164,11 @@ private void publishResults(SignatureScannerReport signatureScannerReport) {
                 SignatureScanPath scanPath = createScanPath(path, maxDepth, signatureScannerExclusionNamePatterns, providedExclusionPatterns);
                 signatureScanPaths.add(scanPath);
             }
-        } else if (dockerTarFile != null) {
-            SignatureScanPath scanPath = createScanPath(dockerTarFile.getCanonicalFile().toPath(), maxDepth, signatureScannerExclusionNamePatterns, providedExclusionPatterns);
+        } else if (dockerTargetData.getSquashedImage().isPresent()) {
+            SignatureScanPath scanPath = createScanPath(dockerTargetData.getSquashedImage().get().getCanonicalFile().toPath(), maxDepth, signatureScannerExclusionNamePatterns, providedExclusionPatterns);
+            signatureScanPaths.add(scanPath);
+        } else if (dockerTargetData.getProvidedImageTar().isPresent()) {
```

Comment:
Don't we want to scan both the squashed and the tar? Will only one be present at a time?

Reply:
same situation as for the binary scanner, except that for sig scan the squashed image is optimal